File: 14.longest-common-prefix.rb
#
# @lc app=leetcode id=14 lang=ruby
#
# [14] Longest Common Prefix
#

# @lc code=start
# @param {String[]} strs
# @return {String}
def longest_common_prefix(strs)
  result = ''

  strs[0].chars.each_with_index do |c, i|
    strs[1..].each do |str|
      return result unless str[i] == c
    end
    result += c
  end

  result
end
# @lc code=end

File: 14.longest-common-prefix.go
/*
 * @lc app=leetcode id=14 lang=golang
 *
 * [14] Longest Common Prefix
 */

package main

import "fmt"

// @lc code=start
func longestCommonPrefix(strs []string) string {
	var prefix string

	for i := 0; i < len(strs[0]); i++ {
		cToCompare := strs[0][i]
		for j := 1; j < len(strs); j++ {
			targetStr := strs[j]
			if i > len(targetStr)-1 {
				return prefix
			}
			if targetStr[i] != cToCompare {
				return prefix
			}
		}

		prefix += string(cToCompare)
	}

	return prefix
}

// @lc code=end

func main() {
	fmt.Println(longestCommonPrefix([]string{"ab", "a"}))
}
